store/web3.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  VuexAction as Action,
+  VuexMutation as Mutation,
+  VuexModule,
+  Module,
+} from "nuxt-property-decorator"
+import WalletConnectProvider from "@walletconnect/web3-provider"
+import Web3modal from "web3modal"
+import { ethers } from "ethers"
+
+const providerOptions = {
+  walletconnect: {
+    package: WalletConnectProvider,
+    options: { infuraId: "3ce35c3d389a4461bffd073fbf27d23e" },
+  },
+}
+
+let web3Modal: Web3modal
+let currentProvider: ethers.providers.Web3Provider
+
+function initializeModal() {
+  // window must be available so we delay instantiating till later
+  if (!web3Modal)
+    web3Modal = new Web3modal({
+      network: "kovan",
+      cacheProvider: false,
+      providerOptions,
+    })
+  return web3Modal
+}
+
+export const getCurrentProvider = () => {
+  return currentProvider
+}
+
+@Module({ stateFactory: true, name: "web3", namespaced: true })
```

Comment:
Just out of curiosity: What does `stateFactory: true` do ?

Reply:
The default is to make the state variable an object, this makes it a function that returns an object.

More on that here:
https://github.com/championswimmer/vuex-module-decorators#module-re-use-use-with-nuxtjs